src/FamilyHubs.Referral.Web/Pages/ProfessionalReferral/LocalOfferDetail.cshtml.cs

```diff
@@ -40,26 +40,30 @@ public async Task<IActionResult> OnGetAsync(string id, string serviceid)
         ReturnUrl = Request.Headers["Referer"].ToString();
         LocalOffer = await _organisationClientService.GetLocalOfferById(serviceid);
         Name = LocalOffer.Name;
-        if (LocalOffer.Locations.Count != 0) ExtractAddressParts(LocalOffer.Locations.First());
+        if (LocalOffer.Locations != null && LocalOffer.Locations.Any()) ExtractAddressParts(LocalOffer.Locations.First());
         GetContactDetails();
 
         return Page();
     }
 
+    public IActionResult OnPost(string id, string serviceId, string name)
+    {
+        return RedirectToPage("/ProfessionalReferral/Safeguarding", new
+        {
+            id = serviceId,
+            name
+        });
+
+    }
+
     public string GetDeliveryMethodsAsString(ICollection<ServiceDeliveryDto>? serviceDeliveries)
     {
         var result = string.Empty;
 
         if (serviceDeliveries == null || serviceDeliveries.Count == 0)
             return result;
 
-        foreach (var name in serviceDeliveries.Select(serviceDelivery => serviceDelivery.Name))
-        {
-            result += result +
-                    (!string.IsNullOrWhiteSpace(name.AsString(EnumFormat.Description)) ?
-                    name.AsString(EnumFormat.Description) + "," :
-                    string.Empty);
-        }
+        result = string.Join(',', serviceDeliveries.Select(serviceDelivery => serviceDelivery.Name).ToArray());
 
         //Remove last comma if present
         if (result.EndsWith(","))

```

Comment:
Now that it's using `string.Join`, this isn't required anymore.

Reply:
Removed